packages/typespec-ts/test/modularUnit/scenarios/multipart/file.md

```diff
@@ -0,0 +1,187 @@
+# Basic file part
+
+```tsp
+model RequestBody {
+  basicFile: HttpPart<File>;
+}
+
+op doThing(@header contentType: "multipart/form-data", @multipartBody bodyParam: RequestBody): void;
+```
+
+## Models
+
+This basic case uses TypeSpec's `Http.File`, which specifies an optional `filename` and `contentType`. Since both are optional, the customer can pass the file's content directly to the `basicFile` property. If the customer wants to specify the filename or content type, they can use the wrapper object.
+
+```ts models
+import {
+  FileContents,
+  createFilePartDescriptor,
+} from "../static-helpers/multipartHelpers.js";
+
+/** model interface RequestBody */
+export interface RequestBody {
+  basicFile:
+    | FileContents
+    | { contents: FileContents; contentType?: string; filename?: string };
+}
+
+export function requestBodySerializer(item: RequestBody): any {
+  return [createFilePartDescriptor("basicFile", item["basicFile"], undefined)];
```

Comment:
wonder why we need to pass undefined here ? I think the third parameter of  createFilePartDescriptor is optional?